File: clients/csharp/examples/tables-alter-type-add.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Tables;
using DataStax.AstraDB.DataApi.Utils;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );

    // Add fields to a user-defined type
    await database.AlterTypeAsync(
      "member",
      new AlterTypeAddFields(
        new()
        {
          ["email"] = DataAPIType.Text(),
          ["credits"] = DataAPIType.Int(),
        }
      )
    );
  }
}

File: clients/java/examples/tables-alter-type-add.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.commands.AlterTypeAddFields;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // Add fields to a user-defined type
    database.alterType(
        "member", new AlterTypeAddFields().addFieldText("email").addFieldInt("credits"));
  }
}
